src/gtc/gtcpp/gtcpp_codegen.py

```diff
@@ -97,12 +97,24 @@ def visit_BuiltInLiteral(self, builtin: BuiltInLiteral, **kwargs: Any) -> str:
     Literal = as_mako("static_cast<${dtype}>(${value})")
 
     def visit_NativeFunction(self, func: NativeFunction, **kwargs: Any) -> str:
-        if func == NativeFunction.SQRT:
-            return "gridtools::math::sqrt"
-        elif func == NativeFunction.MIN:
+        if func == NativeFunction.ABS:
+            return "gridtools::math::abs"
+        if func == NativeFunction.MIN:
             return "gridtools::math::min"
-        elif func == NativeFunction.MAX:
+        if func == NativeFunction.MAX:
             return "gridtools::math::max"
+        if func == NativeFunction.MOD:
+            return "gridtools::math::fmod"
+        if func == NativeFunction.SQRT:
+            return "gridtools::math::sqrt"
+        if func == NativeFunction.POW:
+            return "gridtools::math::pow"
+        if func == NativeFunction.EXP:
+            return "gridtools::math::exp"
+        if func == NativeFunction.LOG:
+            return "gridtools::math::log"
+        if func == NativeFunction.TRUNC:
+            return "gridtools::math::trunc"
```

Comment:
Wouldn't it be easier to replace this set of ifs by a template with a dict lookup for the name of the functions?
Something like:
```
NATIVE_FUNCTION_TO_GT = {
    NativeFunction.ABS: "gridtools::math::sqrt",
    NativeFunction.MIN: "gridtools::math::min",
    ...
}

NativeFunction = as_fmt("{NATIVE_FUNCTION_TO_GT[func]}")
```

Reply:
This gives a less readable error message to the user, so probably not a good idea on its own. But I could easily replace the ifs by a dict lookup with the visit_NativeFunction call, I don’t care.

Reply:
I guess the final user should never see any direct message from the code generator unless there is a bug in the toolchain, so it shouldn't be a problem. However, it's true that for developers working in the toolchain debug could be easier with a explicit visit_ function. I suggest then to use the dict inside the visit_ function to reduce the boilerplate code while still being developer-friendly.

Reply:
Hmm, not sure in this case. The native functions are supported everywhere but in the code generator, so it seems to be the natural place to raise the exception. But of course in the end product it would be nice if all backends supported all frontend features… I replaced the ifs by a dict lookup for now.